PowerPointLabs/PowerPointLabs/SpeechEngine/TextToSpeech.cs

```diff
@@ -59,11 +68,106 @@ public static void SpeakString(String textToSpeak)
             PromptToAudio.Speak(builder);
         }
 
+        private static void SaveStringToWaveFileWithHumanVoice(string textToSave, string filePath)
+        {
+            string accessToken;
+            string textToSpeak = GetHumanSpeakNotesForText(textToSave);
+
+            try
+            {
+                Authentication auth = Authentication.GetInstance();
+                accessToken = auth.GetAccessToken();
+                Logger.Log("Token: " + accessToken);
+            }
+            catch (Exception ex)
+            {
+                Logger.Log("Failed authentication.");
+                Logger.Log(ex.ToString());
+                Logger.Log(ex.Message);
+                return;
+            }
+            string requestUri = UserAccount.GetInstance().GetUri();
+            var cortana = new Synthesize();
+
+            cortana.OnAudioAvailable += SaveAudioToWaveFile;
+            cortana.OnError += ErrorHandler;
+

```

Comment:
Upon audio is available, `SaveAudioToWaveFile` function will be called, which reads in stream input byte by byte, and finally convert to wav file. 
Any error (most likely network error) that occurred during the audio generation process is handled by function `ErrorHandler`.